home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/security/PermissionsIT.java

```diff
@@ -0,0 +1,768 @@
+            case "com.softserveinc.ita.homeproject.client.model.UpdateCooperation":
+                return updateCooperation();
+            case "com.softserveinc.ita.homeproject.client.model.CreateCooperation":
+                return createBaseCooperation();
+            case "com.softserveinc.ita.homeproject.client.model.CreateApartmentInvitation":
+                return createApartmentInvitation();
+            case "com.softserveinc.ita.homeproject.client.model.UpdateApartmentInvitation":
+                return updateApartmentInvitation();
+            case "com.softserveinc.ita.homeproject.client.model.InvitationToken":
+                return getInvitationToken();
+            default:
+                throw new UnsupportedOperationException();
+        }
+    }
+
+    private int getStatusCode(Function<ApiClient, ApiResponse<?>> action, ApiClient unauthorizedClient) {
+        int statusCode;
+        ApiResponse<?> resp = action.apply(unauthorizedClient);
+        statusCode = resp.getStatusCode();
+        return statusCode;
+    }
+
+    public void checkUser(boolean role, int statusCode) {
+        if (role) {
+            Assertions.assertNotEquals(Response.Status.UNAUTHORIZED.getStatusCode(), statusCode);
+            Assertions.assertNotEquals(Response.Status.FORBIDDEN.getStatusCode(), statusCode);
+        } else {
+            if (statusCode == 401) {
+                Assertions.assertEquals(Response.Status.UNAUTHORIZED.getStatusCode(), statusCode);
+            } else {
+                Assertions.assertEquals(Response.Status.FORBIDDEN.getStatusCode(), statusCode);
+            }
+        }
+    }
+
+    @Data
+    @NoArgsConstructor
+    @AllArgsConstructor
+    private static class ApiClientMethods {
+        private Object api;
+        private Method method;
+        private String methodName;
+        private Boolean admin;
+        private Boolean coopAdmin;
+        private Boolean owner;
+        private Boolean userGuest;
+    }
+
+    // populate of class instance via inner class
+    private static List<ApiClientMethods> getAllApiClientMethods(List<Object> clientApiInstances) {
+        ArrayList<ApiClientMethods> allMethods = new ArrayList<>();
+        for (Object instance : clientApiInstances) {
+            getApiMethods(instance);
+            for (Method aMethod : getApiMethods(instance)) {
+                String methodName = aMethod.toString()
+                        .split("\\(")[0]
+                        .split("\\.")[aMethod.toString().split("\\(")[0].split("\\.").length - 1];
+                boolean[] permission = findPermission(methodName.replaceAll("\\QWithHttpInfo\\E", ""));
+                allMethods.add(new ApiClientMethods(instance, aMethod, methodName,
+                        permission[0], permission[1], permission[2], permission[3]));
+            }
+        }
+        return allMethods;
+    }
+
+    private static boolean[] findPermission(String methodName) {
```

Comment:
Yes, if the configuration of permissions for a method changes, then the method must be placed in the block that corresponds to the new configuration of permissions, otherwise the tests will fail.